crates/language_models/src/provider/mistral.rs

```diff
@@ -444,6 +444,35 @@ pub fn into_mistral(
         }
     }
 
+    // The Mistral API requires that tool messages be followed by assistant messages,
+    // not user messages. When we have a tool->user sequence in the conversation,
+    // we need to insert a placeholder assistant message to maintain proper conversation
+    // flow and prevent API errors. This is a Mistral-specific requirement that differs
+    // from other language model APIs.
+    let messages = {
+        let mut fixed_messages = Vec::with_capacity(messages.len());
+        let mut messages_iter = messages.into_iter().peekable();
+
+        while let Some(message) = messages_iter.next() {
+            let is_tool_message = matches!(message, mistral::RequestMessage::Tool { .. });
+            fixed_messages.push(message);
+
+            // Insert assistant message between tool and user messages
+            if is_tool_message {
+                if let Some(next_msg) = messages_iter.peek() {
+                    if matches!(next_msg, mistral::RequestMessage::User { .. }) {
+                        fixed_messages.push(mistral::RequestMessage::Assistant {
+                            content: Some("Continue.".to_string()),
```

Comment:
Does this message have to be "Continue"? Mistral docs seem to pass an empty string: https://docs.mistral.ai/capabilities/function_calling/#parallel_tool_calls

Reply:
I tried with empty string and it seems to give error. I saw couple of threads on other repos where they mentioned you can bypass this by passing " ". But I haven't really tried that give me sometime I can try a string with space to see if that works. Idea behind Continue was to use something which has less noise and gets accepted. I will get back to you after testing this.

Reply:
Tested this @osyvokon with " " it works fine but with "" we get below error. I have made the changes to use " " for now. Let me know that make sense to you.

```
Error interacting with language model
Failed to connect to Mistral API: 400 Bad Request {"object":"error","message":"Assistant message must have either content or tool_calls, but not none.","type":"invalid_request_error","param":null,"code":null}
```